pkg/assets/types.go

```diff
@@ -21,6 +21,12 @@ type BasicAuthCredentials struct {
 	Password string
 }
 
+// AuthorizationCredentials represents a credential to be used with
+// authorization header http authentication, see https://tools.ietf.org/html/rfc7617.
+type AuthorizationCredentials struct {
```

Comment:
why not `type AuthorizationCredentials string`?

Reply:
Makes sense :)